source/agora/consensus/protocol/Nominator.d

```diff
@@ -465,6 +474,51 @@ extern(D):
             log.trace("SCP indicated invalid envelope: {}", scpPrettify(&envelope));
     }
 
+    /***************************************************************************
+
+        Called when a new Block Signature is received from the network.
+
+        Params:
+            block_sig = the SCP envelope
+
+        Returns:
+            true if the SCP protocol accepted this envelope
+
+    ***************************************************************************/
+
+    public void receiveBlockSignature (scope ref const(ValidatorBlockSig) block_sig) @trusted
+    {
+        const cur_height = this.ledger.getBlockHeight();
+        log.trace("Store signature for slot #{} (ledger is at height #{}) for node {}",
+                block_sig.height, cur_height, block_sig.public_key);
+        const Point K = Point(block_sig.public_key[]);
+        this.slot_sigs[block_sig.height][K] = block_sig.signature;
```

Comment:
And what happened to validating the signature from the network? Don't trust data from the network, always validate.

That tells me there's a missing test when someone sends a bad signature through this API.

Reply:
Added validation check. Will add unit test later.